src/components/LineChart/LineChart.jsx

```diff
@@ -0,0 +1,462 @@
+import _ from 'lodash';
+import React from 'react';
+import { lucidClassNames } from '../../util/style-helpers';
+import { createClass } from '../../util/component-types';
+import {
+	minByFields,
+	maxByFields,
+	maxByFieldsStacked,
+	formatDate,
+} from '../../util/chart-helpers';
+import d3Scale from 'd3-scale';
+
+import Axis from '../Axis/Axis';
+import AxisLabel from '../AxisLabel/AxisLabel';
+import Lines from '../Lines/Lines';
+import Points from '../Points/Points';
+
+const cx = lucidClassNames.bind('&-LineChart');
+
+const {
+	arrayOf,
+	func,
+	instanceOf,
+	number,
+	object,
+	shape,
+	string,
+	bool,
+} = React.PropTypes;
+
+/**
+ * {"categories": ["visualizations", "charts"]}
+ *
+ * The line chart presents data over time. Currently only dates are supported
+ * on the x axis and numeric values on the y. If you need discrete values on
+ * the x axis, consider using the `BarChart` instead.
+ */
+const LineChart = createClass({
+	displayName: 'LineChart',
+
+	propTypes: {
+		/**
+		 * Appended to the component-specific class names set on the root element.
+		 */
+		className: string,
+		/**
+		 * Height of the chart.
+		 */
+		height: number,
+		/**
+		 * Width of the chart.
+		 */
+		width: number,
+		/**
+		 * An object defining the margins of the chart. These margins will contain
+		 * the axis and labels.
+		 */
+		margin: shape({
+			top: number,
+			right: number,
+			bottom: number,
+			left: number,
+		}),
+		/**
+		 * Data for the chart. E.g.
+		 *
+		 *     [
+		 *       { x: new Date('2015-01-01') , y: 1 } ,
+		 *       { x: new Date('2015-01-02') , y: 2 } ,
+		 *       { x: new Date('2015-01-03') , y: 3 } ,
+		 *       { x: new Date('2015-01-04') , y: 2 } ,
+		 *       { x: new Date('2015-01-05') , y: 5 } ,
+		 *     ]
+		 */
+		data: arrayOf(object).isRequired,
```

Comment:
should data really be required? what if the screen is loading and there is no data yet, but they still want to render an empty chart?